pyoozie/tags.py

```diff
@@ -0,0 +1,319 @@
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+from __future__ import unicode_literals
+from abc import ABCMeta, abstractmethod
+import re
+import yattag
+
+
+MAX_IDENTIFIER_LENGTH = 39
+REGEX_IDENTIFIER = r'^[a-zA-Z_][\-_a-zA-Z0-9]{0,38}$'
```

Comment:
Not with the power of ["raw strings"](https://docs.python.org/3.4/reference/lexical_analysis.html#string-and-bytes-literals) 😄 

> Both string and bytes literals may optionally be prefixed with a letter 'r' or 'R'; such strings are called raw strings and treat backslashes as literal characters.